test/OpenTelemetry.Instrumentation.AspNetCore.Tests/RouteTests/RoutingTests.cs

```diff
@@ -0,0 +1,195 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+#nullable enable
+
+using System.Diagnostics;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using OpenTelemetry;
+using OpenTelemetry.Metrics;
+using OpenTelemetry.Trace;
+using RouteTests.TestApplication;
+using Xunit;
+using static OpenTelemetry.Internal.HttpSemanticConventionHelper;
+
+namespace RouteTests;
+
+public class RoutingTests : IClassFixture<RoutingTestFixture>
+{
+    private const string OldHttpStatusCode = "http.status_code";
+    private const string OldHttpMethod = "http.method";
+    private const string HttpStatusCode = "http.response.status_code";
+    private const string HttpMethod = "http.request.method";
+    private const string HttpRoute = "http.route";
+
+    private readonly RoutingTestFixture fixture;
+    private readonly List<Activity> exportedActivities = new();
+    private readonly List<Metric> exportedMetrics = new();
+
+    public RoutingTests(RoutingTestFixture fixture)
+    {
+        this.fixture = fixture;
+    }
+
+    public static IEnumerable<object[]> TestData => RoutingTestCases.GetTestCases();
+
+    [Theory]
+    [MemberData(nameof(TestData))]
+    public async Task TestHttpRoute(RoutingTestCases.TestCase testCase, bool useLegacyConventions)
+    {
+        var configuration = new ConfigurationBuilder()
+            .AddInMemoryCollection(new Dictionary<string, string?> { [SemanticConventionOptInKeyName] = useLegacyConventions ? null : "http" })
+            .Build();
+
+        using var tracerProvider = Sdk.CreateTracerProviderBuilder()
+            .ConfigureServices(services => services.AddSingleton<IConfiguration>(configuration))
+            .AddAspNetCoreInstrumentation()
+            .AddInMemoryExporter(this.exportedActivities)
+            .Build()!;
+
+        using var meterProvider = Sdk.CreateMeterProviderBuilder()
+            .ConfigureServices(services => services.AddSingleton<IConfiguration>(configuration))
+            .AddAspNetCoreInstrumentation()
+            .AddInMemoryExporter(this.exportedMetrics)
+            .Build()!;
+
+        await this.fixture.MakeRequest(testCase.TestApplicationScenario, testCase.Path);
+
+        for (var i = 0; i < 10; i++)
+        {
+            if (this.exportedActivities.Count > 0)
+            {
+                break;
+            }
+
+            await Task.Delay(TimeSpan.FromSeconds(1)).ConfigureAwait(false);
+        }
+
+        meterProvider.ForceFlush();
+
+        Assert.Single(this.exportedActivities);
```

Comment:
Later on is this line: ` var activity = this.exportedActivities[0];`

Could be `var activity = Assert.Single(this.exportedActivities);`?

Reply:
!! I never paid attention to the fact that Assert.Single returned the item